json-reporter.js

```diff
@@ -0,0 +1,30 @@
+class MyCustomReporter {
+  constructor(globalConfig, options) {
+    this._globalConfig = globalConfig;
+    this._options = options;
+  }
+
+  onRunComplete(contexts, results) {
+    results.testResults[0].testResults.forEach((item) => {
+      if (item.status !== "failed") {
+        return;
+      }
+      const newLine = "%0A";
+      const name = results.testResults[0].testFilePath;
+
+      if (!item.location) {
+        throw new Error(
+          "You need to invoked jest with argument --testLocationInResults"
+        );
+      }
+
+      const line = item.location.line + 1;
+      const col = item.location.column;
+      const message = item.failureMessages[0].replace(/\n/g, newLine);
+
+      console.log(`::error file=${name},line=${line},col=${col}::${message}`);
```

Comment:
The GitHub runner has some build in matchers and therefore all we have to do is producing output in this format https://github.com/actions/toolkit/issues/260#issuecomment-572377059